cfme/tests/test_rest.py

```diff
@@ -239,6 +252,59 @@ def automation_requests_data():
     } for index in range(1, 5)]
 
 
+@pytest.fixture(scope='function')
+def roles(request, rest_api):
+    if "create" not in rest_api.collections.roles.action.all:
+        pytest.skip("Create roles action is not implemented in this version")
+
+    roles_data = [{
+        "name": "role_name_{}".format(fauxfactory.gen_alphanumeric())
+    } for index in range(1, 5)]
+
+    roles = rest_api.collections.roles.action.create(*roles_data)
+    for role in roles:
+        wait_for(
+            lambda: rest_api.collections.roles.find_by(name=role.name),
+            num_sec=180,
+            delay=10,
+        )
+
+    @request.addfinalizer
+    def _finished():
+        # Magic number 14 is the number of defalt set roles in cfme
+        roles = [r for r in rest_api.collections.roles if r.id > 14]
+        if len(roles) != 0:
+            rest_api.collections.roles.action.delete(*roles)
+
+    return roles
+
+
+@pytest.fixture(scope='function')
+def categories(request, rest_api):
+    if "create" not in rest_api.collections.categories.action.all:
+        pytest.skip("Create categories action is not implemented in this version")
+    ctg_data = [{
+        'name': 'test_category_{}_{}'.format(fauxfactory.gen_alphanumeric().lower(), _index),
+        'description': 'test_category_{}_{}'.format(fauxfactory.gen_alphanumeric().lower(), _index)
+    } for _index in range(0, 5)]
+    ctgs = rest_api.collections.categories.action.create(*ctg_data)
+    for ctg in ctgs:
+        wait_for(
+            lambda: rest_api.collections.categories.find_by(description=ctg.description),
+            num_sec=180,
+            delay=10,
+        )
+
+    @request.addfinalizer
+    def _finished():
```

Comment:
Done, also for roles. 
However I change the filtering by specific prefix to ids of created categories. We need this work around to filtered out the categories/roles which had been deleted in test.